### PR TITLE
Fix t3c-apply default hostname to be short

### LINKED ISSUE #6174: t3c: Status API fails because FQDN is used as a hostname
Depending how the system is configured, a call to `os.HostName()` might return the FQDN or the short name of the server. The status API requires a short hostname.

## Current behavior:
```
ERROR: t3c-request.go:59: 2021-09-08T13:59:45.362466293Z: writing data: getting server 'cdn1cdedge0002.coxlab.net' update status: getting server update status: getting uncached: getting server update status from Traffic Ops 'REDACTED': Traffic Ops returned no update statuses for that server
```

## Expected behavior:
Should use the short hostname. This is usually done using `hostname -s`

## Steps to reproduce:
Make sure `hostname` returns the full FQDN. Run t3c-apply and observe the failure.

### PR DESCRIPTION
**Pull request by rob05c**
Fixes #6174 - t3c-apply with no hostname flag works even if the OS hostname returns a full FQDN rather than a short host name.

Includes tests.
Includes changelog.
No docs, no interface change.

## Which Traffic Control components are affected by this PR?
- Traffic Control Cache Config (T3C, formerly ORT)

## What is the best way to verify this PR?
Run t3c integration tests. Run t3c-apply without the hostname flag, on a machine which returns a full FQDN for `hostname`, verify t3c-apply runs as-expected and generates config for the short hostname of the machine.

## If this is a bugfix, which Traffic Control versions contained the bug?
- 6.0
- 5.1
- 5.0

## PR submission checklist
- [x] This PR has tests 
- [x] This PR has documentation 
- [x] This PR has a CHANGELOG.md entry 
- [x] This PR **DOES NOT FIX A SERIOUS SECURITY VULNERABILITY** (see [the Apache Software Foundation's security guidelines](https://apache.org/security) for details)

### CODE REVIEW

File: cache-config/testing/ort-tests/t3c-os-hostname-short_test.go [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+		tcdata.Divisions, tcdata.Regions, tcdata.PhysLocations,
+		tcdata.CacheGroups, tcdata.Servers, tcdata.Topologies,
+		tcdata.DeliveryServices}, func() {
+		doTestT3cApplyOSHostnameShort(t)
+	})
+}
+
+func doTestT3cApplyOSHostnameShort(t *testing.T) {
+
+	t.Logf("DEBUG TestT3cApplyOSHostnameShort starting")
```
Comment (ocket8888) [inline cache-config/testing/ort-tests/t3c-os-hostname-short_test.go:38]: Was this left in from debugging?

**Review by ocket8888 — COMMENTED** [review]
There are a bunch of `t.Logf("DEBUG ...")` lines in the tests - are those intentional, or left over from debugging?

File: cache-config/testing/ort-tests/t3c-os-hostname-short_test.go (hunk 1 above)
Comment (rob05c) [reply cache-config/testing/ort-tests/t3c-os-hostname-short_test.go:38]: They were intentional, I think they're useful for debugging the test if it breaks. But you're right, I should remove the "DEBUG" text and make them conform to the existing t3c test idioms

Comment (rob05c) [reply cache-config/testing/ort-tests/t3c-os-hostname-short_test.go]: Done

**Review by ocket8888 — APPROVED** [review]
This LGTM but I won't merge until @smalenfant can verify it solves the issue he was seeing in his environment. In which case he may also just merge it without any further involvement from me, which is also fine.

**Review by smalenfant — APPROVED** [review]

**Pull request merged**